pandas/core/indexes/datetimelike.py

```diff
@@ -421,15 +421,25 @@ def _isnan(self):
         return (self.asi8 == iNaT)
 
     @property
-    def asobject(self):
```

Comment:
I actually want to completely remove ``_asobject``, this just adds even more technical debt.

Reply:
Jeff, is it used in `astype` for both `DatetimeIndex`, `PeriodIndex` and `TimedeltaIndex`, so it is fine to have it like this, otherwise this line would be duplicated in all three of them.

Reply:
> so it is fine to have it like this, 

> is it used in astype for both DatetimeIndex, PeriodIndex and TimedeltaIndex,

and that should be addressed as well.

no its not. much better to fix this. otherwise this is just kicking the can.

Reply:
If this method is *only* used there, what is the problem? Then it is a perfect example of a common private helper method that our classes are full of. 

> much better to fix this.

The please indicate what you mean with "fix"